src/frontend/components/FuelCalculator/fuelCalculations.ts

```diff
@@ -0,0 +1,112 @@
+/**
+ * Utility functions for fuel calculations
+ */
+
+import type { FuelLapData } from './types';
+
+/**
+ * Validate if a lap's fuel data is reasonable (outlier detection)
+ */
+export function validateLapData(
+  fuelUsed: number,
+  lapTime: number,
+  recentLaps: FuelLapData[]
+): boolean {
+  // Basic validation
+  if (fuelUsed <= 0 || lapTime <= 0) return false;
+
+  // Need at least 3 laps for statistical validation
+  if (recentLaps.length < 3) return true;
+
+  // Statistical outlier detection (3 standard deviations)
+  const fuelValues = recentLaps.map((l) => l.fuelUsed);
+  const mean = fuelValues.reduce((a, b) => a + b) / fuelValues.length;
+  const variance =
+    fuelValues.reduce((sq, n) => sq + Math.pow(n - mean, 2), 0) /
+    fuelValues.length;
+  const stdDev = Math.sqrt(variance);
+
+  // Allow up to 3 standard deviations from mean
+  return Math.abs(fuelUsed - mean) <= 3 * stdDev;
+}
+
+/**
+ * Calculate weighted average fuel consumption
+ * Recent laps are weighted more heavily than older laps
+ */
+export function calculateWeightedAverage(laps: FuelLapData[]): number {
+  if (laps.length === 0) return 0;
+
+  let weightedSum = 0;
+  let weightSum = 0;
+
+  // Use exponential weighting: more recent laps have higher weight
+  laps.forEach((lap, idx) => {
+    const weight = Math.pow(1.2, laps.length - idx - 1);
+    weightedSum += lap.fuelUsed * weight;
+    weightSum += weight;
+  });
+
+  return weightedSum / weightSum;
+}
+
+/**
+ * Convert liters to gallons
+ */
+export function litersToGallons(liters: number): number {
+  return liters * 0.264172;
+}
+
+/**
+ * Convert gallons to liters
+ */
+export function gallonsToLiters(gallons: number): number {
+  return gallons / 0.264172;
+}
+
+/**
+ * Format fuel amount for display
+ */
+export function formatFuel(
+  liters: number,
+  units: 'L' | 'gal',
+  decimals: number = 2
+): string {
+  const value = units === 'gal' ? litersToGallons(liters) : liters;
+  return `${value.toFixed(decimals)} ${units}`;
+}
+
+/**
+ * Detect if a lap crossing occurred
+ * Lap crossing happens when distance percentage goes from high (>0.9) to low (<0.1)
+ */
+export function detectLapCrossing(
+  currentDistPct: number,
+  lastDistPct: number
+): boolean {
+  return lastDistPct > 0.9 && currentDistPct < 0.1;
+}
+
+/**
+ * Check if current session flags indicate green flag conditions
+ */
+export function isGreenFlag(sessionFlags: number): boolean {
+  const YELLOW_FLAG = 0x00004000; // 1 << 14
+  const CAUTION_FLAG = 0x00008000; // 1 << 15
+  const RED_FLAG = 0x00010000; // 1 << 16
```

Comment:
Not important but it might be good to put these under `src/types` 

All the flags are defined in irsdk_defines.h but that's in C++

Reply:
> Not important but it might be good to put these under `src/types`
> 
Do you want a `flags.ts` in `src/types`?
<img width="182" height="222" alt="image" src="https://github.com/user-attachments/assets/aa0faae8-3b29-4271-9558-5aa7389cc6cf" />